examples/StreamLayerReadExample.cpp

```diff
@@ -0,0 +1,231 @@
+  client_settings.authentication_settings = auth_settings;
+  client_settings.network_request_handler = std::move(http_client);
+
+  // Create subscription, used kSerial or kParallel subscription mode
+  olp::dataservice::read::SubscribeRequest subscribe_request;
+  subscribe_request.WithSubscriptionMode(subscription_mode);
+
+  // set options, default values described here:
+  // https://developer.here.com/olp/documentation/data-api/api-reference-stream.html
+  olp::dataservice::read::ConsumerOptions expected_options = {
+      olp::dataservice::read::ConsumerOption("auto.commit.interval.ms",
+                                             int32_t{5000}),
+      olp::dataservice::read::ConsumerOption("auto.offset.reset", "earliest"),
+      olp::dataservice::read::ConsumerOption("enable.auto.commit", "false"),
+      olp::dataservice::read::ConsumerOption("fetch.max.bytes",
+                                             int32_t{52428800}),
+      olp::dataservice::read::ConsumerOption("fetch.max.wait.ms", int32_t{500}),
+      olp::dataservice::read::ConsumerOption("fetch.min.bytes", int32_t{1}),
+      olp::dataservice::read::ConsumerOption("group.id", "group_id_1"),
+      olp::dataservice::read::ConsumerOption("max.partition.fetch.bytes",
+                                             int32_t{1048576}),
+      olp::dataservice::read::ConsumerOption("max.poll.records", int32_t{500})};
+  subscribe_request.WithConsumerProperties(
+      olp::dataservice::read::ConsumerProperties(expected_options));
+
+  auto read_from_stream_layer = [&]() {
+    // Create stream layer client with settings and catalog, layer specified
+    olp::dataservice::read::StreamLayerClient client(olp::client::HRN{catalog},
+                                                     layer_id, client_settings);
+    if (!CreateSubscription(client, subscribe_request)) {
+      return false;
+    }
+    RunPoll(client);
+    if (!DeleteSubscription(client)) {
+      return false;
+    }
+    return true;
+  };
+
+  int result = 0;
+  if (subscription_mode ==
+      olp::dataservice::read::SubscribeRequest::SubscriptionMode::kParallel) {
+    // With parallel subscription you can read large volumes of data in a
+    // parallel manner. The subscription and message reading workflow is similar
+    // to serial subscription except that you are allowed to create multiple
+    // subscriptions for the same aid, catalog, layer, group.id combination
+    // using multiple processes/threads. This allows you to read and commit
+    // messages for each subscription in parallel.
+    OLP_SDK_LOG_INFO_F(kLogTag, "Run Poll in parallel, threads count = %u",
+                       kNumberOfThreads);
+    std::atomic<int> value(0);
+    std::thread threads[kNumberOfThreads];
+    for (auto& thread : threads) {
+      thread = std::thread([&]() {
+        if (!read_from_stream_layer()) {
+          // acumulate result from all threads
+          value.store(-1);
+        }
+      });
```

Comment:
It is very confusing that you use lambda in lambda. Why dont you directly set the atomic in read_from_stream_layer()?